src/libraries/System.Runtime.InteropServices/tests/Ancillary.Interop/IUnmanagedVirtualMethodTableProvider.cs

```diff
@@ -27,20 +27,53 @@ public void Deconstruct(out IntPtr thisPointer, out ReadOnlySpan<IntPtr> virtual
         }
     }
 
-    public interface IUnmanagedVirtualMethodTableProvider<T> where T : IEquatable<T>
+    public unsafe interface IUnmanagedVirtualMethodTableProvider<T> where T : IEquatable<T>
     {
         protected VirtualMethodTableInfo GetVirtualMethodTableInfoForKey(T typeKey);
 
         public sealed VirtualMethodTableInfo GetVirtualMethodTableInfoForKey<TUnmanagedInterfaceType>()
-            where TUnmanagedInterfaceType : IUnmanagedInterfaceType<T>
+            where TUnmanagedInterfaceType : IUnmanagedInterfaceType<TUnmanagedInterfaceType, T>
         {
             return GetVirtualMethodTableInfoForKey(TUnmanagedInterfaceType.TypeKey);
         }
-    }
 
+        public static int GetVirtualMethodTableLength<TUnmanagedInterfaceType>()
+            where TUnmanagedInterfaceType : IUnmanagedInterfaceType<TUnmanagedInterfaceType, T>
+        {
+            return TUnmanagedInterfaceType.VirtualMethodTableLength;
+        }
+
+        public static void* GetVirtualMethodTableManagedImplementation<TUnmanagedInterfaceType>()
+            where TUnmanagedInterfaceType : IUnmanagedInterfaceType<TUnmanagedInterfaceType, T>
+        {
+            return TUnmanagedInterfaceType.VirtualMethodTableManagedImplementation;
+        }
+
+        public static void* GetUnmanagedWrapperForObject<TUnmanagedInterfaceType>(TUnmanagedInterfaceType obj)
+            where TUnmanagedInterfaceType : IUnmanagedInterfaceType<TUnmanagedInterfaceType, T>
+        {
+            return TUnmanagedInterfaceType.GetUnmanagedWrapperForObject(obj);
+        }
 
-    public interface IUnmanagedInterfaceType<T> where T : IEquatable<T>
+        public static TUnmanagedInterfaceType GetObjectForUnmanagedWrapper<TUnmanagedInterfaceType>(void* ptr)
+            where TUnmanagedInterfaceType : IUnmanagedInterfaceType<TUnmanagedInterfaceType, T>
+        {
+            return TUnmanagedInterfaceType.GetObjectForUnmanagedWrapper(ptr);
+        }
+    }
+
+    public unsafe interface IUnmanagedInterfaceType<TInterface, TKey>
```

Comment:
Add full docs for all the APIs in here (`Ancillary.Interop`)? Probably also be nice for future-us when proposing/adding the APIs.